dotnet/samples/GettingStartedWithProcesses/Utilities/MermaidRenderer.cs

```diff
@@ -0,0 +1,119 @@
+
+            // Ensure the directory exists
+            string directoryPath = Path.GetDirectoryName(outputFilePath)
+                ?? throw new InvalidOperationException("Could not determine the directory path.");
+            if (!Directory.Exists(directoryPath))
+            {
+                throw new DirectoryNotFoundException($"The directory '{directoryPath}' does not exist.");
+            }
+        }
+        else
+        {
+            // Use the assembly's directory for relative paths
+            string? assemblyPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            if (assemblyPath == null)
+            {
+                throw new InvalidOperationException("Could not determine the assembly path.");
+            }
+
+            string outputPath = Path.Combine(assemblyPath, "output");
+            Directory.CreateDirectory(outputPath); // Ensure output directory exists
+            outputFilePath = Path.Combine(outputPath, filenameOrPath);
+        }
+
+        // Download Chromium if it hasn't been installed yet
+        BrowserFetcher browserFetcher = new();
+        browserFetcher.Browser = SupportedBrowser.Chrome;
+        await browserFetcher.DownloadAsync();
+
+        // Define the HTML template with Mermaid.js CDN
+        string htmlContent = $@"
+        <html>
+            <head>
+                <style>
+                    body {{
+                        display: flex;
+                        align-items: center;
+                        justify-content: center;
+                        margin: 0;
+                        height: 100vh;
+                    }}
+                </style>
+                <script type=""module"">
+                    import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';
+                    mermaid.initialize({{ startOnLoad: true }});
+                </script>
+            </head>
+            <body>
+                <div class=""mermaid"">
+                    {mermaidCode}
+                </div>
+            </body>
+        </html>";
+
+        // Create a temporary HTML file with the Mermaid code
+        string tempHtmlFile = Path.Combine(Path.GetTempPath(), "mermaid_temp.html");
+        try
+        {
+            await File.WriteAllTextAsync(tempHtmlFile, htmlContent);
+
+            // Launch Puppeteer-Sharp with a headless browser to render the Mermaid diagram
+            using (var browser = await Puppeteer.LaunchAsync(new LaunchOptions { Headless = true }))
+            using (var page = await browser.NewPageAsync())
+            {
+                await page.GoToAsync($"file://{tempHtmlFile}");
+                await page.WaitForSelectorAsync(".mermaid"); // Wait for Mermaid to render
+                await page.ScreenshotAsync(outputFilePath, new ScreenshotOptions { FullPage = true });
+            }
+        }
+        catch (Exception ex)
+        {
+            throw;
```

Comment:
throw a specific exception with a meaningful message that included the exception thrown. Perhaps include any known exceptions thrown by puppeteer something like :

``` c#
catch (IOException ex)  
{  
    throw new IOException("An error occurred while accessing the file.", ex);  
}  
catch (Exception ex) // Catch any other exceptions that might occur  
{  
    throw new InvalidOperationException("An unexpected error occurred during Mermaid diagram rendering.", ex);  
} 
```